client/packages/common/src/ui/forms/JsonForms/components/DateTime.tsx

```diff
@@ -45,11 +49,17 @@ export const datetimeTester = rankWith(5, isDateTimeControl);
 
 const UIComponent = (props: ControlProps) => {
   const [error, setError] = React.useState('');
-  const { data, handleChange, label, path } = props;
+  const { data, handleChange, label, path, uischema } = props;
+  if (!props.visible) {
+    return null;
+  }
+  const dateFormatter = useFormatDateTime().customDate;
```

Comment:
The hook always needs to be called otherwise React might complain, so the visible check has to be after the hook...

The check is there twice as well...

Reply:
Oops -- removed duplicate: de9c267a0